userspace/libpman/include/libpman.h

```diff
@@ -275,33 +280,31 @@ extern "C"
 	 * @brief Instrument the bpf probe with the right sc_set. This API
 	 * sets both interesting syscalls and interesting tracepoints.
 	 *
-	 * @param sc_set pointer to the interesting sc_set 
+	 * @param sc_set pointer to the interesting sc_set
 	 *
 	 * @return `0` on success, `1` in case of error.
 	 */
-	int pman_enforce_sc_set(bool *sc_set);
+	int pman_enforce_sc_set(bool* sc_set);
 
 	/**
 	 * @brief Receive a pointer to `struct scap_stats` and fill it
 	 * with info about the number of events and number of drops.
 	 *
-	 * @param scap_stats_struct opaque pointer to `struct scap_stats`.
-	 * We used an opaque pointer because we don't want to introduce scap
-	 * definitions in this file.
+	 * @param scap_stats_struct pointer to `struct scap_stats`.
 	 * @return `0` on success, `errno` in case of error.
 	 */
-	int pman_get_scap_stats(void* scap_stats_struct);
+	int pman_get_scap_stats(struct scap_stats* scap_stats_struct);
 
 	/**
-	 * @brief Get scap_stats_v2 structure filled with the statistics.
+	 * @brief Retutn a `scap_stats_v2` struct filled with statistics.
 	 *
-	 * @param scap_stats_v2_struct opaque pointer to `struct scap_stats_v2` held in modern_bpf_engine handle
 	 * @param flags holding statistics category flags.
-	 * @param nstats Pointer reflecting number of statistics in returned buffer.
+	 * @param nstats number of stats allocated.
+	 * @param rc return code, SCAP_FAILURE in case of error.
 	 *
-	 * @return `0` on success, `errno` in case of error.
+	 * @return pointer to `struct scap_stats_v2`
 	 */
-	int pman_get_scap_stats_v2(void* scap_stats_v2_struct, uint32_t flags, uint32_t* nstats);
+	struct scap_stats_v2* pman_get_scap_stats_v2(uint32_t flags, uint32_t* nstats, int32_t* rc);
```

Comment:
Hmm we now change this again? Didn't we discuss in the initial scap stats v2 PR that we want to return the error code? Either way is fine with me just wanted to point out that it initially was similar to how you changed it and then based on reviewers comments changed it.

Reply:
that's true you are right, in the end, this approach seems more simple to manage in the code so I would go for it! If anyone has something against it I can change it again of course

Reply:
I am fine with leaving as is :)